File: internal/model/llm_provider.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type LLMProvider struct {
	ID        bson.ObjectID `json:"id" bson:"_id,omitempty"`
	Name      string        `json:"name" bson:"name"`
	BaseURL   string        `json:"base_url" bson:"base_url"`
	APIKey    string        `json:"api_key" bson:"api_key"`
	Models    []string      `json:"models" bson:"models"`
	CreatedAt time.Time     `json:"created_at" bson:"created_at"`
	UpdatedAt time.Time     `json:"updated_at" bson:"updated_at"`
}

File: web/src/api/llm_provider.ts
import request from './request'

export interface LLMProvider {
  id: string
  name: string
  base_url: string
  api_key: string
  models: string[]
  created_at: string
  updated_at: string
}

export const llmProviderApi = {
  list: () => request.get('/llm-providers'),
  get: (id: string) => request.get(`/llm-providers/${id}`),
  create: (data: Partial<LLMProvider>) => request.post('/llm-providers', data),
  update: (id: string, data: Partial<LLMProvider>) => request.put(`/llm-providers/${id}`, data),
  delete: (id: string) => request.delete(`/llm-providers/${id}`),
}
